src/internal/log/loggers.go

```diff
@@ -57,6 +64,41 @@ func SetGRPCLogLevel(l zapcore.Level) {
 	grpcLevel.SetLevel(l)
 }
 
+type atomicLeveler interface {
+	Level() zapcore.Level
+	SetLevel(zapcore.Level)
+}
+
+// revertLogLevel sets up a log level change that is reverted to an original level after the
+// duration, logging the revert with the text of msg, ensuring that even with multiple concurrent
+// calls, the most recent call cancels the actions of prior calls.
+func revertLogLevel(tPtr *atomic.Pointer[time.Timer], levelVar atomicLeveler, originalLevel, newLevel zapcore.Level, d time.Duration, msg string) {
+	wasSet := make(chan struct{})
+	oldTimer := tPtr.Swap(time.AfterFunc(d, func() {
+		<-wasSet
+		cur := levelVar.Level()
+		levelVar.SetLevel(originalLevel)
+		zap.L().Info(msg, zap.Stringer("from", cur), zap.Stringer("to", originalLevel))
```

Comment:
I think this should be like max level, always print no matter what.  Otherwise this becomes difficult to reason about.  

The user set's it to something above info and now we can't see the changes.  Use of the global logger isn't great.

Reply:
That makes sense to me.  That's actually what we do when changing it.  I used Info here because Error logs alarm people and this isn't an error, but on the other hand, changing the log level to Error also isn't an error but that's what it's logged at.

Reply:
I changed the API to let the caller pass in the logging function to use; this way, the logs also correspond to the request-id that caused the level to be changed.